2021.11.0 (#20137)
home-assistant/home-assistant.github.io: source/_posts/2021-11-03-release-202111.markdown

date: 2021-11-03 00:00:00
date_formatted: "November 3, 2021"
author: Franck Nijhof
author_twitter: frenck
comments: true
categories:
- Release-Notes
- Core
og_image: /images/blog/2021-11/social.png
---

<a href='/integrations/#version/2021.11'><img src='/images/blog/2021-11/social.png' style='border: 0;box-shadow: none;'></a>

Point eleven (`.11`), which means it is November! 🥶 (or 🏖 depending on where
you are in this world, of course.)

It also means: This is the second last release of the year...

Hacktoberfest is over! In the month October, people
contribute to projects (like Home Assistant) to celebrate open source.
I've seen lots of new faces and new contributors this month. To those I want
to say: Welcome! Thanks for contributing your free spare time; Thanks for
being a hero! ❤️

And, there is more! Something that isn't really visible. In the background,
an army of contributors are improving a tremendous amount of Home
Assistant code, which improves stability, readability, and maintainability  of
our project's codebase. As a matter of fact, because of this incredible work,

> counted by the number of contributions, this release might just a well be one

Reviewer: ```suggestion
counted by the number of contributions, this release might just as well be one
```

Screenshot of WLED that now has a button that links to the WLED web interface of the device directly.
</p>

If an integration provides a single web interface to configure multiple devices,
it can even deep-link to the specific device page for that device.

Over **50!** integrations have already added support for this feature, which is
fantastic! Just to name a few integrations: MQTT, Tasmota, WLED, Synology, Plex,
Netatmo, Pi-hole, RainMachine, Plugwise, Sure Petcare, Axis, UniFi Controller,
FRITZ!Box, and many more!

If you are a custom integration developer, you can add support for this feature
too; [Read all about it on our developer blog](https://developers.home-assistant.io/blog/2021/10/26/config-entity#device-configuration-url).

## Entity categorization

Since we introduced the concept of devices, we have been seeing more and
more entities being created, which is great! It means we can automate on a lot
more device features. However, this does impose other issues, like:
clutter on a generated dashboard, unused entities exposed to voice assistants,
or turn on/off things in areas as a side-effect. Let me try to explain the
latter issue with an example.

> Let consider a wallplug. It can be turned on/off using a switch entity,

Reviewer: ```suggestion
Let's consider a wall plug. It can be turned on/off using a switch entity,
```

> however, my wallplug also has a switch to enable a child lock that controls

Reviewer: ```suggestion
however, my wall plug also has a switch to enable a child lock that controls
```

- Add support for fan groups! Thanks [@bdraco]!
- Thanks to [@ludeeus], the Supervisor integration now has binary sensors that
  can tell you which add-on is running or not.
- [@allenporter] blew our minds this release by adding initial support for
  WebRTC streams and cameras to Home Assistant.
- But had a good reason, he added support for Nest Battery Cameras
  and Nest battery Doorbell Cameras to Home Assistant! Thanks [@allenporter]!
- When you log in to Home Assistant, there is now a "Keep me logged in"
  checkbox! Awesome [@MartinTuroci](http://github.com/MartinTuroci)
- [@goyney](https://github.com/goyney) updated the Material Design Icon to
  version v6.4.95 ensuring we have all the latest icons available in our new
  beautiful icon picker.
- Sure PetCare now supports the Felaqua devices, thanks [@benleb]!
- The MQTT number platform now has support for providing a unit of measurement
  and resetting the number field (by [@frenck] and [@emontnemery]).
- The Broadlink integration added sensors for voltage, current, overload,
  and total consumption. Thanks [@felipediel]!
- WLED now has the much requested sliders for speed & intensity controls.
  Thanks, [@rytilahti]!

> - OpenGarage has seens lots of updates and love from [@Danielhiversen], adding

Reviewer: ```suggestion
- OpenGarage has seen lots of updates and love from [@Danielhiversen], adding
```

  lots of sensors in the process.
- With the recent improvements made to the VLC media player integration and
  add-on, there is now a way to get TTS working by connecting some speakers to
  your Blue, ODROID or Raspberry Pi. Thanks, [@mib1185] & [@MartinHjelmare]!

Adding support for long-term statistics is a much requested feature; This
release adds support for long-term statistics to the following integrations:

- AEMET (thanks [@Noltari])
- Efery (thanks [@tkdrob])
- EnOcean (thanks [@davidlb])
- Gogogate2 (thanks [@thomas-svrts])
- Huisbaasje (thanks [@dennisschroer])
- Nest (thanks [@allenporter])
- Ondilo ICO (thanks [@TheNogl])
- OpenWeatherMap (thanks [@dkt01])
- Plugwise (thanks [@bouwew])
- REST (thanks [@koying])
- Scrape sensor (thanks [@lukas-hetzenecker])
- Tado (thanks [@Noltari])
- Tado sensors (thanks [#58111])
- WAQI (thanks [@davidlb])

## New Integrations

We welcome the following new integration this release:

- [LOOKin][lookin docs], done by [@ANMalko]

## Integrations now available to set up from the UI

> The following integrationsare now available via the Home Assistant UI:

Reviewer: ```suggestion
The following integrations are now available via the Home Assistant UI:
```

based on entities (unlike areas) and thus still work as before.

([@emontnemery] - [#57671])

{% enddetails %}

{% details "Amazon Alexa / Google Assistant" %}

Entities that are now shown on the device page on the `configuration` or `diagnostic`
section, will **no longer be exposed** by default to Amazon Alexa or Google Assistant.

If you want to expose these entities, you'll need to expose them explicitly.

([@emontnemery])

{% enddetails %}

{% details "Recorder / Databases" %}

The Home Assistant project has now explicitly defined a set of databases
that are supported by the recorder.

We now support the following databases:

- MariaDB ≥ 10.3
- MySQL ≥ 8.0
- PostgreSQL ≥ 12
- SQLite ≥ 3.32.1

All other databases and database versions are now deprecated.

If your used or configured database doesn't match any of the above requirements,
a warning will be logged. As of Home Assistant 2022.1, the recorder will
no longer connect to an unsupported database.

([@emontnemery] - [#58161]) ([recorder docs])

---

It is no longer possible to filter out `state_changed` events from the
recorder's `event_types` exclude filter.

> A warning will be logged in Home Assistant Core 2022.11 and the configuration

Reviewer: ```suggestion
A warning will be logged in Home Assistant Core 2021.11 and the configuration
```

will be rejected in Home Assistant Core 2022.2.

{% enddetails %}

{% details "Python 3.8: Deprecation warning" %}

Python 3.8 has been deprecated because Python 3.10 has been released.
Support will be dropped in Home Assistant 2022.1.

Support for Python 3.10 hasn't completed yet, however we aim to complete
that before the 3.8 deprecation period has ended. It is advised to upgrade
to Python 3.9 at this point.

If you run Home Assistant OS, Supervised or the Container installation type:

> No action is required from your end, we will take of this for you.

Reviewer: ```suggestion
No action is required from your end, we will take care of this for you.
```

([@balloob] - [#57079])

{% enddetails %}

{% details "Aurora ABB Solar PV" %}

The Aurora ABB Solar PV integration migrated to configuration via the UI.
Configuring Aurora ABB Solar PV via YAML configuration has been deprecated
and will be removed in a future Home Assistant release.

Your existing YAML configuration is automatically imported on upgrade to this
release; and thus can be safely removed from your YAML configuration after upgrading.

([@davet2001] - [#36300]) ([aurora_abb_powerone docs])

{% enddetails %}

{% details "AVM FRITZ!Box Tools" %}

> AVM FRITZ!Box Tools services now ask for a `device_id` and no more for a `entity_id`.

Reviewer: ```suggestion
AVM FRITZ!Box Tools services now ask for a `device_id` and no longer for an `entity_id`.
```

> Please review your automation and scripts accordly.

Reviewer: ```suggestion
Please review your automation and scripts accordingly.
```

Configuring Environment Canada via YAML configuration has now been deprecated and will be
removed in a future Home Assistant release.

Your existing YAML configuration is automatically imported on upgrade to this release;
and thus can be safely removed from your YAML configuration after upgrading.

Additionally, non-loop radar images are no longer supported; Sensor entities
now have the location of the station as a part of the sensor's entity name.

([@gwww] - [#57127]) ([environment_canada docs])

{% enddetails %}

{% details "Flux LED/MagicLight" %}

The Flux LED integration has been migrated to configuration via the UI.
Configuring Flux LED via YAML configuration has now been deprecated and will be
removed in a future Home Assistant release.

Your existing YAML configuration is automatically imported on upgrade to this release;
and thus can be safely removed from your YAML configuration after upgrading.

([@icemanch] - [#56354]) ([flux_led docs])

{% enddetails %}

{% details "HomeKit" %}

The previously deprecated HomeKit configuration options `safe_mode`,
`zeroconf_default_interface`, and `auto_start` have now been removed.

> If stil have these options in your YAML configuraion, you must be remove them

Reviewer: ```suggestion
If you still have these options in your YAML configuration, you must be remove them
```

([@peternijssen] - [#57974]) ([spider docs])

{% enddetails %}

{% details "Microsoft Text-to-Speech (TTS)" %}

The default voice is changed to `JennyNeural`; The previous default `ZiraRUS`
(Standard Voice) is no longer supported in eastus or any other region that
has rolled out GA neural voices as Standard voices become deprecated.

([@tylergibson] - [#58499]) ([microsoft docs])

{% enddetails %}

{% details "OctoPrint" %}

The OctoPrint integration migrated to configuration via the UI. Configuring
OctoPrint via YAML configuration has been deprecated and will be removed in
a future Home Assistant release.

Your existing YAML configuration is automatically imported on upgrade to this
release; and thus can be safely removed from your YAML configuration after upgrading.

The following sensors have been removed:

- `sensor.<printer>_time_remaining`
- `sensor.<printer>_time_elapsed`

The following sensors have been added:

- `sensor.octoprint_start_time`
- `sensor.octoprint_estimated_finish_time`

([@rfleming71] - [#58040]) ([octoprint docs])

{% enddetails %}

{% details "OpenGarage" %}

The attributes (`wifi_signal ` and `distance_sensor`) for OpenGarage covers

> are removed and replaced by individual sensors sensors.

Reviewer: ```suggestion
are removed and replaced by individual sensors.
```

> Also, `door_state ` is removed as an attribute, but you can use the state

Reviewer: ```suggestion
Also, `door_state` is removed as an attribute, but you can use the state
```

- `stop_program`
- `stop_zone`

([@bachya] - [#57145]) ([rainmachine docs])

{% enddetails %}

{% details "Renault" %}

The `last_update` state attribute has been moved to a stand-alone entity
(and dropped from battery and location entities). If you used this attribute
in your scripts or automations; you'll need to adjust those to this change.

([@epenet] - [#57750]) ([renault docs])

{% enddetails %}

{% details "SimpliSafe" %}

SimpliSafe requires a new web-based authentication method. When starting up,
Home Assistant will automatically prompt you to re-authorize your instance.

([@bachya] - [#57212]) ([simplisafe docs])

{% enddetails %}

{% details "SolarEdge" %}

The SolarEdge integration now provides unique IDs for its entities, so the
entities are now visible and customizable in the Home Assistant frontend.

Some names have been corrected as they had incorrect descriptions. This may
cause some of your entity IDs provided by SolarEdge to have changed this release.

In case you have been using these entities in your automations, scripts, or
dashboards, you can either rename those entity IDs to the old entity ID
or adjust the places you've used them accordingly.

> Additionally, if sensor you used disappeared, it might have been disabled by

Reviewer: ```suggestion
Additionally, if a sensor you used disappeared, it might have been disabled by
```

> default during upgrade. Please be sure to check the if the entity is

Reviewer: ```suggestion
default during upgrade. Please be sure to check if the entity is
```

enabled in that case.

([@terminet85] - [#55902]) - ([@purcell-lab] - [#58773])

{% enddetails %}

{% details "Sonos" %}

The `sonos.set_option` service has been removed and replaced with
individual `switch` entities which both control the features and display
if currently enabled.

The `night_sound` and `speech_enhance` attributes on the
Sonos `media_player` entities are also removed.

> If you are using those in your automation or scripts, you'll need to replacet

Reviewer: ```suggestion
If you are using those in your automations or scripts, you'll need to replace
```

> hese with the new `switch` entities.

Reviewer: ```suggestion
these with the new `switch` entities.
```

([@jjlawren] - [#54502]) ([sonos docs])

{% enddetails %}

{% details "Stookalert" %}

The Stookalert integration has been migrated to configuration via the UI.
Configuring Stookalert via YAML configuration has now been deprecated and will be
removed in a future Home Assistant release.

Your existing YAML configuration is automatically imported on upgrade to this release;
and thus can be safely removed from your YAML configuration after upgrading.

([@frenck] - [#57119]) ([stookalert docs])

{% enddetails %}

{% details "Supervisor" %}

The 2021.09 release deprecated the usage of "snapshot" in the Supervisor (`hassio`)
integration and was scheduled to be removed in 2021.11, that is now.

Old services and service data that contained "snapshot" has now been removed,

> you need to update your configuration if you use any of these to continue to use it.

Reviewer: ```suggestion
and you need to update your configuration if you use any of these to continue to use it.
```

release; and thus can be safely removed from your YAML configuration after upgrading.

([@MartinHjelmare] - [#57513]) ([vlc_telnet docs])

{% enddetails %}

{% details "Xiaomi Miio" %}

Attributes from the vacuum entities have been replaced with sensors.

This means that all your references to these attributes must be replaced with
its sensor in your automations, scripts and dashboards.

Some sensors have been disabled by default and need manual enablement,
once the integration has been set up. For example, the attribute
`mop_attached: true` has now been replaced with
`binary_sensor.<NAME>_mop_attached`.

([@OGKevin] - [#54990]) ([xiaomi_miio docs])

{% enddetails %}

{% details "Yeelight" %}

The option to override the Yeelight model will only appear if the detected
model is not in the database. Otherwise, it will be hidden.

([@bdraco] - [#56967]) ([yeelight docs])

{% enddetails %}

## Farewell to the following

The following integrations are no longer available as of this release:

- **Essent** has been removed. The API is no longer usable for this integration
  and has therefore been removed. ([@TheLastProject] - [#56991])

> - **Lyft** has been previously deprecated and now removed. API key are no longer

Reviewer: ```suggestion
- **Lyft** has been previously deprecated and now removed. API keys are no longer
```
